### PR TITLE
feat: Update Password Facade (GH-1162)

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-1162

### CODE REVIEW

File: projects/core/src/user/occ/user.service.spec.ts [hunk 1]
```diff
@@ -256,4 +257,28 @@ describe('OccUserService', () => {
...
+
+  describe('update password: ', () => {
+    it('should update the password for userId', () => {
+      const userId = 'test@test.com';
+      const oldPassword = 'OldPass123!';
+      const newPassword = 'NewPass456!';
+
+      service
+        .updatePassword(userId, oldPassword, newPassword)
+        .subscribe(result => expect(result).toEqual(''));
```
Comment (znikola) [inline projects/core/src/user/occ/user.service.spec.ts]: Maybe move the `expect` out of the `subscribe` block?

File: projects/core/src/user/occ/user.service.spec.ts [hunk 2]
```diff
@@ -256,4 +257,28 @@ describe('OccUserService', () => {
...
+
+      const mockReq = httpMock.expectOne(req => {
+        console.log(req);
+        console.log('ser body: ', req.serializeBody());
```
Comment (znikola) [inline projects/core/src/user/occ/user.service.spec.ts]: Please remove `console.log`s from lines `272` and `273`

File: projects/core/src/user/occ/user.service.spec.ts (hunk 1 above)
Comment (plabadie) [reply projects/core/src/user/occ/user.service.spec.ts]: Agreed, the `expect` should not be in the `subscribe` block.

Fixed.

File: projects/core/src/user/occ/user.service.spec.ts (hunk 2 above)
Comment (plabadie) [reply projects/core/src/user/occ/user.service.spec.ts]: Fixed.

File: projects/core/src/user/facade/user.service.ts [hunk 3]
```diff
@@ -1,26 +1,34 @@
...
 } from '../../occ/occ-models/index';
+import {
+  getProcessErrorFactory,
+  getProcessLoadingFactory,
+  getProcessSuccessFactory,
+} from '../../process';
+import * as fromProcessStore from '../../process/store/process-state';
 import { UserRegisterFormData } from '../model/user.model';
+import * as fromStore from '../store/index';
+import { UPDATE_PASSWORD_PROCESS_ID } from '../store/index';
```
[the remark this answers is not in the document]
Comment (plabadie) [reply projects/core/src/user/facade/user.service.ts]: Fixed.

File: projects/core/src/user/occ/user.service.ts [hunk 4]
```diff
@@ -191,7 +191,11 @@ export class OccUserService {
     return this.occEndpoints.getEndpoint(USER_ENDPOINT);
   }
 
-  updatePassword(userId: string, oldPassword: string, newPassword: string) {
+  updatePassword(
+    userId: string,
+    oldPassword: string,
+    newPassword: string
+  ): Observable<{}> {
```
Comment (znikola) [inline projects/core/src/user/occ/user.service.ts]: ❤️ type safety

**Pull request merged**